src/parser/GenericParser.test.ts

```diff
@@ -0,0 +1,281 @@
+        ]);
+
+        const findFilesAsyncMock = findFilesAsync as jest.Mocked<typeof findFilesAsync>;
+        findFilesAsyncMock.mockImplementation((config) => {
+            return mockedFindTwoFiles();
+        });
+
+        const treeParserSpied = jest
+            .spyOn(TreeParser, "parse")
+            .mockImplementation((file, assumedLanguage) => {
+                return Promise.resolve({
+                    fileExtension: file.fileExtension,
+                    filePath: file.filePath,
+                    language: Language.CPlusPlus,
+                    tree: tree,
+                });
+            });
+
+        const calculateMetricsSpied = jest
+            .spyOn(MetricCalculator.prototype, "calculateMetrics")
+            .mockImplementation(async (parsedFilePromise) => [
+                (await parsedFilePromise).filePath,
+                expectedFileMetricsMap,
+            ]);
+
+        const parser = new GenericParser(getParserConfiguration("clearly/invalid"));
+        const actualResult = await parser.calculateMetrics();
+
+        expect(actualResult.fileMetrics).toEqual(
+            new Map([
+                ["clearly/invalid/path1.cc", expectedFileMetricsMap],
+                ["clearly/invalid/path2.cpp", expectedFileMetricsMap],
+            ])
+        );
+        expect(calculateMetricsSpied).toHaveBeenCalledTimes(2);
+        expect(treeParserSpied).toHaveBeenCalledTimes(2);
+    });
+
+    it("should return the appropriate value and call the appropriate functions in case there is an error while parsing the tree", async () => {
+        const expectedFileMetricsMap = new Map([
+            ["ERROR", { metricName: "ERROR", metricValue: -1 }],
+        ]);
+
+        const findFilesAsyncMock = findFilesAsync as jest.Mocked<typeof findFilesAsync>;
+        findFilesAsyncMock.mockImplementation((config) => {
+            return mockedFindFilesAsync();
+        });
+
+        jest.spyOn(TreeParser, "parse").mockImplementation(async (file, assumedLanguage) => {
+            throw new Error("Baaaaaah");
+        });
+
+        jest.spyOn(MetricCalculator.prototype, "calculateMetrics").mockResolvedValue([
+            "clearly/invalid/path.cpp",
+            expectedFileMetricsMap,
+        ]);
+
+        const errorSpy = jest.spyOn(console, "error").mockImplementation(() => {
+            /* Do nothing */
+        });
+
+        const parser = new GenericParser(getParserConfiguration("clearly/invalid/path.cpp"));
+        const actualResult = await parser.calculateMetrics();
+
+        // TODO: do no longer include errors into the list of results, use "info" field instead.
+        expect(actualResult.fileMetrics).toEqual(
+            new Map([["clearly/invalid/path.cpp", expectedFileMetricsMap]])
+        );
+        expect(errorSpy).toHaveBeenCalled();
+    });
+
+    it("should return the appropriate value and call the appropriate functions in case there is an error while calculating metrics", async () => {
```

Comment:
Let's try to be more explicit: What is an appropriate value? What it the appropariate action/function that is called? Which type of error is thrown?